Sources/MapboxNavigation/NavigationMapView.swift

```diff
@@ -2120,12 +2120,60 @@ open class NavigationMapView: UIView {
         if let selected = waypointTest?.first {
```

Comment:
This method becomes more complex with time, I wonder if we could cover it with unit-tests to make sure that delegate methods are called based on expected priority.

Reply:
Right now it still follows the previous sequence, but between the waypoints and route, it would check the duration annotation. I'll add test for this one later.